### DISCUSSION TITLE
pcb for hub75 matrix and lolin D32 pro

### DISCUSSION THREAD

**Discussion by heidepiek** [PlummersSoftwareLLC/NightDriverStrip#483] [Ideas]
With easyeda tried to design a pcb.  First time, so I thought I'd show it here on the github page and maybe someone will have some tips and suggestions.  Routed the pinout exactly like my working mesmerizer matrix and esp. Accounted for extra gpio pins!

Greetzz harrie

[SCH_schematic lolin D32 hub75 shield_2023-10-31.json](https://github.com/PlummersSoftwareLLC/NightDriverStrip/files/13220457/SCH_schematic.lolin.D32.hub75.shield_2023-10-31.json)
[PCB_PCB_schematic lolin D32 hub75 shield_2023-10-31.json](https://github.com/PlummersSoftwareLLC/NightDriverStrip/files/13220458/PCB_PCB_schematic.lolin.D32.hub75.shield_2023-10-31.json)

**Comment by robertlipe** [comment]
Nice.

If you're going for low quantities/personal use, consider an S3 with fast
flash and ram. Save the cost of the uart and use the onboard serial/USB.
Makes tossing around larger screens easier.

**Reply by heidepiek** [reply]
Thanks robert, it is indeed for personal use. This dummy is trying to understand a bit what you are telling me. "Save the cost of the uart and use the onboard serial/USB" You mean power the esp with the usb port?  Have tried several shields and they work well with e.g. 64x64 matrices. You have to use thicker traces on the pcb then though. The matrices use a lot of amps! My intention was to use a thick 20 amp supply on the pcb to feed the matrix directly (hence the two thick connectors) And also feed the esp32 via the 5volt line. Two birds with one stone. 
The pcb I designed is not ready yet. Still needs some tweaking. I also plan to make a connection for the IR receiver. I also got that working tonight. Had to change some ir codes, because I don't have the same remote that dave recommended. And "consider an S3 with fast flash and ram" Hmm I thought the lolin d32 pro with specs 16mb flash and 8mb ram was pretty spacious anyway, or am I thinking wrong? Trying to follow you guys a bit, but you are of a different caliber than me! 😏

**Comment by robertlipe** [comment]
Ah, I see now you're mounting an existing dev board and not a module. So
you're paying for the serial chip anyway. I though you were doing like
Mesmerizer and going from the module to the HUB75.

If you (or in your case, the Lolin board) were using the ESP32-S3 module
instead of the ESP32-S module, you wouldn't need th esreial chip; the
ESP32-S3 has a USB host controller and CDC/ACM on-board, so you get JTAG +
serial straight from pins 18 and 19 (USB-D+ and D-) and don't need a CH340
or CH912x or whatever. YOu can make a very functional S3 design with almost
<https://www.hackster.io/news/minimizing-an-esp32-s3-development-board-down-to-the-module-itself-e6170256de30>
no support componentry. See "QT py" or " XIAO ESP32-S3" for examples,
though I wouldn't go quite that minimal.

I've never tried to power a matrix through the board with an ESP32 on it.
As you observed, they're power-hungry. I've always run wires straight to
the supply.  I've never even used the screw terminals on the official
mesmerizer board as I'm skeptical that the 3A it could possibly get form
the USB-C jack at 5V just seemed too wimpy.

I don't have EasyEDA, but remember to bring out JTAG (if you don't go
S3...which doesn't need more pins) as well as pins for analog + digital
audio in, remote analog in, and SPI/I2C in case you want to add random
sensors or radios or such. You might not want to pay for connectors for
everything, but blank PCB layout real estate is relatively cheap.

If cost isn't a primary concern and you're building a feature-rich project,
the ESP32-S3 is just a super choice. Teaching Mesmerizer's FFT code about
the S3 would be a good holiday project.

Good luck!

**Comment by heidepiek** [comment]
Last night tinkering with easyeda, I think i wil order it like this at jlcpcb. 

![hub75 shield lolin d32 pro](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/925d8de7-0197-4c09-92fc-d00d6359ca1a)
![hub75 shield lolin d32 pro backside](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/8cec1e05-dfdb-4009-802d-ee536cea4f45)

![hub75 shield lolin d32 pro schematic](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/b3aa7b14-968e-4d71-9a2e-e96c26d197a8)

**Comment by heidepiek** [comment]
Got my pcb`s today from jlcpcb. Here the result. I personally think it`s very successful! Modified the holder of dave a bit for the microphone and ir sensor. Are now hidden in the front of the holder. Soldered the few parts. Clicked the Pcb on the hub and used the : default_envs = mesmerizer_lolin build. Does it well. Web server good, remote control works. And it looks neat.
Should there be interest in the gerber files, ask and I'll put a link in this discussion!



![mesmerizer foto weather](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/4f7589fe-e3f0-4a92-8d27-ee3b519942a5)

![front](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/9db11e30-a45c-44d8-957e-c3e3d49d11fd)

![backside pcb mounted](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/16d5bc6a-f991-42f9-a413-88fd8dcf38e1)

![foto 5](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/bdade813-828d-4faa-8eab-2d09eccc9214)

![foto 6](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/029feba5-6e89-449a-9c3d-614e53b6928d)

![foto 7](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/2062b0ba-79b1-4f99-9fd6-fb5a0abf08a0)

![ir and mic 1](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/ffb4fc8a-184f-47d5-b84a-7a8817daa1f7)

![ir and mic 2](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/e08675bb-5f75-49a2-a992-d311bc777404)

Greetz Harrie

**Comment by robertlipe** [comment]
Nicely done. Looks great. That's worth an entry in the "show and tell"
group here for sure.

Do you get any ground hum from the long unshielded, unpaired analog input
lines for the mic?
Could you simplify the hand "routing" of the IR sensor by rotating it 90
degrees on the PCB?

As an adjacent hack, consider a USB-C "trigger board" to get 5V 5A and feed
it (and your MCU) from a commodity USB-C laptop charger. I stock boards
approximately like
https://www.aliexpress.us/item/3256804378476973.html
(not that exact model and I'm not recommending that exact model...) and
have been replacing all my broken/janky barrel jacks and power supplies in
my life with USB-C Power Delivery power supplies.  Be sure to look for one
that can do 5V/5A for this case. Even the 64x64's really only need about 4A
with every pixel on full white (which is like the visual intensity of the
sun...) so they work great here.

Why are you using a module with battery and TFT and other jacks (Stemma?
Qt?) ? Are you using those features (maybe you have room temperature
sensors that you're displaying to Mesmerizer or a side-car TFT screen...)
or are they just along for the ride on the board you're using?

As simple as it is to design with the ESP32 modules, if I were making a
board, I'd probably have LCSC/JLCPCB splash down the module, lay down an
LMS1177 regulator, add buttons for RESET/BOOT, a USB-C jack, route the
GPIOs to HUB75, sensors as you have above, and just go all in. But as an
engineer, I know I'm prone to overkill/overcomplication. But working with
the modules is about as simple as workign with the boards you're buying and
a lot less expensive if you need a couple. He went a little TOO
minimalistic, but this video <https://www.youtube.com/watch?v=tQ0dL_9M1wI>
shows how low you can go.


Having been on the sidelines of watching you bring this up, I'm happy for
you!

RJL

**Reply by heidepiek** [reply]
Thanks robert, why I used this very board is that at first I had trouble getting the mesmerizer to work on the boards I had. When you mentioned wrover and psram I searched the internet. And tinytronics sold this board that I am using now. Just bought it and tried it, and I got it to work. That all that other stuff was on it, well it's on it. But don't use it. Like to work with loose boards. Am not into loose chips myself. Now I can just take it off the board and possibly use it for something else. Love simple. Don't have any interference in the microphone.  That the ir is twisted on the pcb is because I had to put the microphone on the other side, and I had to rotate it. So the other side with the ir also rotated. Symmetrical. Something between the ears! Autistic. Do not plan to add more things. Like it well enough. Will get a nice spot here in my mancave. Have several craft projects that require power,And have a fat 20 amp power supply hanging under the board. That is where the mesmerizer gets power from. So no messing around with plugs and connections. Straight to the power supply. 😉

Harrie

**Comment by davepl** [comment]
Love it!  I really like how you integrated the mic and IR sensor. I’ve been looking for a small panel mount unit to incorporate into my frame models… if anyone knows of a small one, please post a link!

Also neat that you took the approach of integrating with an existing module rather than being a dedicated board like the Mesermizer PCB.

- Dave

**Reply by heidepiek** [reply]
Thanks dave, like to keep it simple. This one will get a nice spot in my mancave. 😉

**Comment by mggates39** [comment]
Nice.
@heidepiek can you pull this branch https://github.com/mggates39/NightDriverStrip/tree/feature/New-Weather-Icons and run the Weather Effect to see if it looks okay with the new icons?

**Reply by heidepiek** [reply]
I had some problems with my pc, so i used my old one from way back. I think there was u bug in platformio. But my other pc is back on line. I downloaded your branch again. And i could build it. 

![new weather icons](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/868e333c-d4ad-41a0-959f-1d6ace7976e5)


Harrie

**Reply by mggates39** [reply]
Thank you!  That looks just like I thought it would.  It has both a night time, with a moon, and a daytime, with a sun weather icon for mostly cloudy. You can see the difference in the color of the clouds too.

**Reply by heidepiek** [reply]
Looks really nice! 👍 😉
